test/e2e/util/util.go

```diff
@@ -1479,6 +1492,97 @@ func EnsurePodsWithEmptyDirPVsHaveSafeToEvictAnnotations(t *testing.T, ctx conte
 	})
 }
 
+func EnsureReadOnlyRootFilesystem(t *testing.T, ctx context.Context, hostClient crclient.Client, hcpNs string) {
+	// By default, we enable readOnlyRootFilesystem in every container.
+	// This testchecks to make sure that every container has this field enabled, unless manually specified in auditedAppContainersNoRORFS
+	t.Run("EnsureReadOnlyRootFilesystem", func(t *testing.T) {
+		g := NewWithT(t)
+
+		hcpPods := &corev1.PodList{}
+		if err := hostClient.List(ctx, hcpPods, &client.ListOptions{
+			Namespace: hcpNs,
+		}); err != nil {
+			t.Fatalf("cannot list hostedControlPlane pods: %v", err)
+		}
+
+		// a list of applications that are allowed to have Pod.Spec.Containers[*].SecurityContext.ReadOnlyRootFilesystem == false
+		// auditedAppContainersNoRORFS[pod.Name][pod.Spec.Containers[*]] indicates that particular container is allowed to be false
+		auditedAppContainersNoRORFS := map[string]map[string]struct{}{}
+
+		for _, pod := range hcpPods.Items {
+			// skip etcd and feature-gate-generator pods
+			// If added to the list of audited pods,it will fail the e2e check on older release branches since e2e is ran from main.
```

Comment:
wouldn't this entire validation fail for older releases since most pods won't have these changes?
I'd think you'd want a check like 
https://github.com/openshift/hypershift/blob/f64e73d54a6d3759b5511d61bc78d58b1b392c53/test/e2e/autoscaling_test.go#L149
at the beginning of the function